src/client/common/application/notebook.ts

```diff
@@ -94,12 +94,19 @@ export class VSCodeNotebook implements IVSCodeNotebook {
     ): Disposable {
         return notebook.registerNotebookContentProvider(notebookType, provider, options);
     }
-    public registerNotebookKernelProvider(
-        selector: NotebookDocumentFilter,
-        provider: NotebookKernelProvider
-    ): Disposable {
-        return notebook.registerNotebookKernelProvider(selector, provider);
+
+    // IANHU
+    // public registerNotebookKernelProvider(
+    // selector: NotebookDocumentFilter,
+    // provider: NotebookKernelProvider
+    // ): Disposable {
+    // return notebook.registerNotebookKernelProvider(selector, provider);
+    // }
+    //createNotebookKernel(options: NotebookKernelOptions): NotebookKernel2;
```

Comment:
Can we not leave the old code in tact?
Or is the push model implementation of kernels complete?

Reply:
No, this was an old half done branch. I think it was just based on that to get the proposed API right.